Src/FluentAssertions/Numeric/DecimalAssertions.cs

```diff
@@ -14,12 +15,12 @@ internal DecimalAssertions(decimal value)
         {
         }
 
-        private protected override decimal? CalculateDifferenceForFailureMessage(decimal expected)
+        private protected override string CalculateDifferenceForFailureMessage(decimal subject, decimal expected)
         {
             try
             {
-                var difference = checked(Subject - expected);
-                return difference != 0 ? difference : null;
+                decimal difference = checked(subject - expected);
+                return difference != 0 ? difference.ToString(CultureInfo.InvariantCulture) : null;
```

Comment:
I'm a bit concerned that we're no longer using the value formatters that are specifically tailored for that and add

Reply:
FYI, we don't do that currently with `Invariant($" (difference of {difference}).")`.

The thing is that we cannot return a difference in type `uint` for two `uints` as the signed difference can be negative.